cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -237,6 +239,25 @@ func Run(c *config.CompletedConfig, stopCh <-chan struct{}) error {
 		}
 		saTokenControllerInitFunc := serviceAccountTokenControllerStarter{rootClientBuilder: rootClientBuilder}.startServiceAccountTokenController
 
+		// start client to resource providers
+		if len(c.ResourceProviderClients) > 0 {
+			controllerContext.ResourceProviderClients = make(map[string]clientset.Interface)
+			controllerContext.ResourceProviderNodeInformers = make(map[string]coreinformers.NodeInformer)
+			for i, rpClient := range c.ResourceProviderClients {
+				rpId := "rp" + strconv.Itoa(i)
+				resourceInformerFactory := informers.NewSharedInformerFactory(rpClient, 0)
+				resourceInformerFactory.Start(controllerContext.Stop)
+				controllerContext.ResourceProviderClients[rpId] = rpClient
+				controllerContext.ResourceProviderNodeInformers[rpId] = resourceInformerFactory.Core().V1().Nodes()
+				klog.V(2).Infof("Created the node informer %p from resource provider %s",
+					controllerContext.ResourceProviderNodeInformers[rpId].Informer(), rpId)
+				go controllerContext.ResourceProviderNodeInformers[rpId].Informer().Run(controllerContext.Stop)
+			}
+		} else {
+			controllerContext.ResourceProviderNodeInformers = make(map[string]coreinformers.NodeInformer, 1)
+			controllerContext.ResourceProviderNodeInformers["0"] = controllerContext.InformerFactory.Core().V1().Nodes()
```

Comment:
is there a reason go informers["0"].Informer().Run(...) not to start in this branch?

Reply:
The else branch is used for single cluster (TP/RP has the same api server). controllerContext.InformerFactory started somewhere else.

> is there a reason go informers["0"].Informer().Run(...) not to start in this branch?